src/Terrain/MapLibrary.java

```diff
@@ -1,36 +1,58 @@
 package Terrain;
 
+import java.util.ArrayDeque;
 import java.util.ArrayList;
 
-import Terrain.Maps.CageMatch;
-import Terrain.Maps.FiringRange;
+import Terrain.MapInfo.MapNode;
 import Terrain.Maps.MapReader;
-import Terrain.Maps.SpannIsland;
 import Terrain.Maps.TestRange;
+import lombok.var;
 
 public class MapLibrary
 {
   private static ArrayList<MapInfo> availableMaps;
+  private static MapNode rootMap;
   
   public static ArrayList<MapInfo> getMapList()
   {
-    if(null == availableMaps)
+    if( null == availableMaps )
     {
       loadMapInfos();
     }
     return availableMaps;
   }
+  public static MapNode getMapGraph()
+  {
+    if( null == rootMap )
+    {
+      loadMapInfos();
+    }
+    return rootMap;
+  }
   
   private static void loadMapInfos()
   {
     availableMaps = new ArrayList<MapInfo>();
     availableMaps.add(TestRange.getMapInfo());
-    availableMaps.add(FiringRange.getMapInfo());
-    availableMaps.add(SpannIsland.getMapInfo());
-    availableMaps.add(CageMatch.getMapInfo());
-    availableMaps.addAll(MapReader.readMapData());
+
+    rootMap = MapReader.readMapData();
```

Comment:
Instead of changing the `MapReader` API and introducing a bunch of loop logic here, why not keep the API the same; let `MapReader` do the directory traversal internally (maybe split into a utility function) and just return the list of maps as before? That way `MapLibrary` is back to just worrying about maps, not where they come from.

Reply:
Because the `rootMap` is the API used by the GUI, returning two separate results from a function isn't convenient, and I don't really want `MapReader` to have static state.